lib/puppet/provider/openldap_access/olc.rb

```diff
@@ -30,7 +30,7 @@ def self.instances
           suffix = line.split[1]
         when %r{^olcAccess: }
           begin
-            position, what, bys = line.match(%r{^olcAccess:\s+\{(\d+)\}to\s+(\S+(?:\s+filter=\S+)?(?:\s+attrs=\S+)?(?:\s+val=\S+)?)(\s+by\s+.*)+$}).captures
+            position, what, bys = line.match(%r{^olcAccess:\s+\{(\d+)\}to\s+(\S+(?:\s+filter=\S+)?(?:\s+attrs=\S+)?(?:\s+val(?:/\S+)?(?:\.\S+)?=\S+)?)(\s+by\s+.*)+$}).captures
```

Comment:
We do not have unit tests to check this, and I fail to match the first part of the change:
1. `(?:/\S+)?` <--- this part
2. `(?:\.\S+)?`

For the second part, it match the `[.<basic-style>]` in [`val[.<basic-style>]=<regex>`](https://www.openldap.org/doc/admin24/access-control.html), where basic-style seems to be a static list that we can use to tighten parsing, i.e. `(?:\.(?:regex|exact))?`.

Do you have an example of what you want to match with the first part of your change?

Reply:
Hi,

we want to match the following `olcAccess` value:

```
{0}to dn.base="" attrs=supportedControl val/objectIdentifierMatch=1.2.840.113556.1.4.319 by * none
```

Reply:
Thank you! Looks like the administrator guide only present a small subset of what the man page talk about.